### PR TITLE
oom: some improvements and fixes

### PR DESCRIPTION
**Pull request by anitazha**

### CODE REVIEW

File: src/oom/oomd.c [hunk 1]
```diff
@@ -142,10 +142,8 @@ static int run(int argc, char *argv[]) {
                 return log_error_errno(r, "Failed to get SwapTotal from /proc/meminfo: %m");
 
         r = safe_atollu(swap, &s);
-        if (r < 0)
-                return log_error_errno(r, "Failed to parse SwapTotal from /proc/meminfo: %s: %m", swap);
-        if (s == 0)
-                return log_error_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Requires swap to operate");
+        if (r < 0 || s == 0)
+                log_warning("Swap is currently not detected; memory pressure usage will be degraded");
```
Comment (bluca) [inline src/oom/oomd.c:146]: Perhaps log_info is enough? It's most likely a conscious configuration decision after all, it's unlikely that swap disappears unintentionally

**Review by bluca — APPROVED** [review]
Just one minor comment, else LGTM!

File: src/oom/oomd.c (hunk 1 above)
Comment (keszybz) [reply src/oom/oomd.c:146]: I think a log_warning is reasoanble. The only case where swap wouldn't be useful would be with a workload that either is overprovisioned or tailored to available memory. But in those cases, oomd is also not needed. In all other cases, at least zram should be used.

File: src/oom/oomd-manager.c [hunk 2]
```diff
@@ -314,12 +317,12 @@ static int monitor_cgroup_contexts_handler(sd_event_source *s, uint64_t usec, vo
...
                 return log_error_errno(r, "Failed to check if memory pressure exceeded limits");
         else if (r == 1) {
-                /* Check if there was reclaim activity in the last interval. The concern is the following case:
+                /* Check if there was reclaim activity in the given interval. The concern is the following case:
                  * Pressure climbed, a lot of high-frequency pages were reclaimed, and we killed the offending
                  * cgroup. Even after this, well-behaved processes will fault in recently resident pages and
                  * this will cause pressure to remain high. Thus if there isn't any reclaim pressure, no need
                  * to kill something (it won't help anyways). */
-                if (oomd_memory_reclaim(m->monitored_mem_pressure_cgroup_contexts)) {
+                if ((usec_now - m->last_reclaim_at) <= RECLAIM_DURATION_USEC) {
```
Comment (keszybz) [inline src/oom/oomd-manager.c:325]: () not needed.

**Review by keszybz — COMMENTED** [review]
LGTM.

**Pull request merged**